1Password/onepassword-sdk-python
# Add python toolchain-script

This MR will create a two step release process that you can run on any branch and release the Python SDK.

This requires GitLab CLI installed and the MakeFile checks if you have it or not.

Also, a PAT will have to be created and exported to GITHUB_CLI_TOKEN environment variable to run the gitlab CLI. The minimum required scopes for the token are: repo, read:org, and gist.

The first script prep-release.sh will update the version/build numbers and create a new changelog for the release

The second script release.sh will create the release branch, add all necessary files, and create the release with the correct changelog and version

This process will ensure that the developer can verify the files before pushing a release.

A README is in the `src/onepassword/changelogs` folder on step by step process on how to push a release.

Was tested on fork of the repo.
Build Number:
![image](https://github.com/1Password/onepassword-sdk-python/assets/103403052/adf89b56-7d18-411f-a168-9a1c9908f583)
Version:
![image](https://github.com/1Password/onepassword-sdk-python/assets/103403052/9655080c-c5c0-47a7-ad32-eb0b173ccc68)

## To Test
1. Go into the test python sdk repo (https://github.com/MOmarMiraj/onepassword-sdk-python). I sent you three all invites to access the repo which should give you the correct permissions.
3. Test on main/feature branch.. run `make prep-release`.. follow instructions
4. Run `make release`

I have updated the fork repo to include the latest changes and edited it so the release will be created on the fork repo not the actual repo.. so just run the scripts and should be good to go.
